server/src/main/java/org/opensearch/index/compositeindex/datacube/MetricStat.java

```diff
@@ -10,29 +10,50 @@
 
 import org.opensearch.common.annotation.ExperimentalApi;
 
+import java.util.Arrays;
+import java.util.List;
+
 /**
  * Supported metric types for composite index
  *
  * @opensearch.experimental
  */
 @ExperimentalApi
 public enum MetricStat {
-    COUNT("count"),
-    AVG("avg"),
-    SUM("sum"),
-    MIN("min"),
-    MAX("max");
+    COUNT("count", null),
+    SUM("sum", null),
+    MIN("min", null),
+    MAX("max", null),
+    AVG("avg", new MetricStat[] { COUNT, SUM });
 
     private final String typeName;
+    private final MetricStat[] baseMetrics;
 
-    MetricStat(String typeName) {
+    MetricStat(String typeName, MetricStat[] baseMetrics) {
         this.typeName = typeName;
+        this.baseMetrics = baseMetrics;
```

Comment:
Check for not empty

Reply:
baseMetrics will be empty for everything except derivedMetric, so skipping for now. Let me know if i'm not understanding your comment correctly